src/main/java/com/example/school/SchoolApplication.java

```diff
@@ -145,23 +151,142 @@ public static void main(String[] args) {
                 .build();
-        var mark2 = Activity.builder()
+        var activity2 = Activity.builder()
                 .lesson(lesson2)
                 .student(student2)
                 .mark(Optional.of(10))
                 .isPresent(true)
                 .build();
-        var mark3 = Activity.builder()
+        var activity3 = Activity.builder()
                 .lesson(lesson3)
                 .student(student7)
                 .mark(Optional.empty())
                 .isPresent(false)
                 .build();
+        var activity4 = Activity.builder()
+                .lesson(lesson1)
+                .student(student2)
+                .mark(Optional.empty())
+                .isPresent(false)
+                .build();
+        var activity5 = Activity.builder()
+                .lesson(lesson3)
+                .student(student2)
+                .mark(Optional.empty())
+                .isPresent(false)
+                .build();
+        var activity6 = Activity.builder()
+                .lesson(lesson3)
+                .student(student6)
+                .mark(Optional.of(9))
+                .isPresent(false)
+                .build();
+        var activity7 = Activity.builder()
+                .lesson(lesson3)
+                .student(student7)
+                .mark(Optional.of(5))
+                .isPresent(true)
+                .build();
+
+        var activities = List.of(activity1, activity2, activity3, activity4, activity5);
+        var activities1 = List.of(activity1, activity2, activity3, activity4, activity5, activity6, activity7);
+        System.out.println(getTheWorstTruantInSchool(activities));
+        System.out.println(getListOfTopGoodStudentsInSchool(activities1));
+    }
+
+    // method that gets a list of students from the school who skipped the most lessons
+    private static List<Student> getTheWorstTruantInSchool(List<Activity> activities) {
+        Map<Student, Integer> amountTruanciesEachStudent = new HashMap<>();
+        for (Activity activity : activities) {
+            int countTruancy = 1;
+            if (!activity.isPresent()) {
+                if (!amountTruanciesEachStudent.containsKey(activity.getStudent())) {
+                    amountTruanciesEachStudent.put(activity.getStudent(), countTruancy);
+                } else {
+                    countTruancy = amountTruanciesEachStudent.get(activity.getStudent());
+                    countTruancy++;
+                    amountTruanciesEachStudent.put(activity.getStudent(), countTruancy);
+                }
+            }
+        }
+        int maxAmountOfTruancies = amountTruanciesEachStudent.values().stream()
+                .max(Integer::compareTo).orElse(-1);
+        return amountTruanciesEachStudent.entrySet().stream()
+                .filter(e -> e.getValue() == maxAmountOfTruancies)
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toList());
+
+    }
```

Comment:
It would be better if for one method, one PR.

Reply:
Also I simplify method a little bit, please change your method according my changes and figure out how it works.

Reply:
`// method that gets a list of students from the school who skipped the most lessons`

   ` private static List<Student> findBestTruantsInSchool(List<Activity> activities) {`
        `var amountSkippedLessonsForEachStudent = countSkippedLessonsForEachStudent(activities);`
       ` int maxAmountOfTruancies = amountSkippedLessonsForEachStudent.values().stream()`
              `  .max(Integer::compareTo)`
              `  .orElse(-1);`
        `return amountSkippedLessonsForEachStudent.entrySet().stream()`
          `      .filter(e -> e.getValue() == maxAmountOfTruancies)`
          `      .map(Map.Entry::getKey)`
          `      .collect(Collectors.toList());`
   ` }`
    `private static Map<Student, Integer> countSkippedLessonsForEachStudent(List<Activity> activities) {`
        `Map<Student, Integer> amountTruanciesEachStudent = new HashMap<>();`
        `for (var activity : activities) {`
           ` if (!activity.isPresent()) {`
                `var student = activity.getStudent();`
                `var skippedLesson = 1;`
                `if(amountTruanciesEachStudent.containsKey(student)) {`
                    `skippedLesson = amountTruanciesEachStudent.get(student);`
               ` }`
               ` amountTruanciesEachStudent.put(student, skippedLesson);`
           ` }`
       ` }`
        `return amountTruanciesEachStudent;`
   ` }`

Reply:
Also it would be better to add sql script which will do the same but according database.

Reply:
skippedLesson = amountTruanciesEachStudent.get(student) +1;